engine/src/main/java/io/zeebe/engine/processor/workflow/handlers/IOMappingHelper.java

```diff
@@ -45,34 +48,56 @@
       variablesState.removeTemporaryVariables(elementInstanceKey);
     }
 
-    if (hasOutputMappings) {
-      mergeTool.reset();
-
-      final DirectBuffer variables =
-          determineVariables(variablesState, elementInstanceKey, outputMappings);
-
-      mergeTool.mergeDocumentStrictly(variables, outputMappings);
-      final DirectBuffer mergedVariables = mergeTool.writeResultToBuffer();
+    if (!outputMappings.isEmpty()) {
```

Comment:
I like to have these kind of variables `hasOutputMappings` the `!` are often overseen